classes/Customer.php

```diff
@@ -1064,13 +1064,14 @@ public static function getCurrentCountry($idCustomer, Cart $cart = null)
      */
     public function toggleStatus()
     {
-        parent::toggleStatus();
+        $statusUpdated = parent::toggleStatus();
+        if ($statusUpdated) {
+            Db::getInstance()->execute('UPDATE `'._DB_PREFIX_.bqSQL($this->def['table']).'`
+                SET `date_upd` = NOW()
+                WHERE `'.bqSQL($this->def['primary']).'` = '.(int) $this->id);
+        }
 
-        /* Change status to active/inactive */
-        return Db::getInstance()->execute('
-        UPDATE `'._DB_PREFIX_.bqSQL($this->def['table']).'`
-        SET `date_upd` = NOW()
-        WHERE `'.bqSQL($this->def['primary']).'` = '.(int) $this->id);
+        return $statusUpdated;
```

Comment:
Good catch! However, I wouldn't lose the result of the `date_upd` update.

Is there a reason you didn't return both results? You could have combined them like this:
```php
    return parent::toggleStatus() && 
            Db::getInstance()->execute('UPDATE `'._DB_PREFIX_.bqSQL($this->def['table']).'`
                SET `date_upd` = NOW()
                WHERE `'.bqSQL($this->def['primary']).'` = '.(int) $this->id);
```

If the `parent::toggleStatus()` returns false, the second request won't be executed.

Reply:
I think that the return value must indicated if the status has changed. 
So if status changed and the second query failed : the status was still updated.

Perhaps the `date_upd` should be updated in the same update query (in `ObjectModelCore` `toggleStatus()` ) ?

Reply:
Actually ... It should be already updated!

`toggleStatus()` calls `update()`, which updates this column.

Reply:
Of course ! 
I've tested and we can delete `toggleStatus` method from `Customer` class !
Do you want me to make the change in other MR and close this one ?

Reply:
I also tested on my side and saw the `date_upd` was still updated.
Yes, you should suggest the change in a PR if you want to get the contribution. :)